web-common/src/Halogen/Monaco.purs

```diff
@@ -215,6 +227,19 @@ handleQuery (GetModelMarkers f) = do
       markers <- Monaco.getModelMarkers monaco model
       pure $ f markers
 
+handleQuery (GetDecorationRange decoratorId f) = do
+  withEditor \editor -> do
+    liftEffect do
+      model <- liftEffect $ Monaco.getModel editor
+      let
+        decoRange = Monaco.getDecorationRange model decoratorId
+      pure $ f decoRange
+
+handleQuery (SetDeltaDecorations first last f) = do
+  withEditor \editor -> do
+    decoId <- liftEffect $ Monaco.setDeltaDecorations editor first last
+    pure (f decoId)
```

Comment:
to be consistent, I think the other `handleQuery` functions should use `pure (something)` rather than `pure $ something`